### DISCUSSION TITLE
Trying to install tasmota on the gosound sp111

### DISCUSSION THREAD

**Discussion by GiannisDorokidis** [arendst/Tasmota#22230] [Q&A]
Im trying to flash an sp111 with tasmota for the first time. First i put the device into flash mode by grounding for a bit gpio0 and then i read the firmware to a bin so i can restore it. Then i erashed the flash it worked. Lastly i tried the web installer for tasmota and it said it had problems. So i tried flashing tasmota-minimal.bin and it didnt work. The cp210 usb to uart was sending data but it wasnt receiving any.
Here is the guide i was following https://www.malachisoord.com/2019/11/24/flashing-custom-firmware-on-a-gosund-sp111
I got the some version of the plug like the pictures.

backup of the default firmware:
```esptool.exe --port COM6 read_flash 0x00000 0x100000 esp2.bin
[21:56]
esptool.exe -p COM6 erase_flash erase flash
[21:56]
esptool.py v4.8.1
Serial port COM6
Connecting...
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting...
Detecting chip type... ESP8266
Chip is ESP8285N08
Features: WiFi, Embedded Flash
Crystal is 26MHz
MAC: 48:55:19:5b:1b:2a
Stub is already running. No upload is necessary.
Erasing flash (this may take a while)...
Chip erase completed successfully in 1.9s
Hard resetting via RTS pin...
[21:57]```
##then tried writing tasmota-minimal on it
```[21:57]
esptool.exe -p COM6 write_flash -fs 1MB -fm dout 0x0 tasmota-minimal.bin
esptool.py v4.8.1
Serial port COM6
Connecting......................................

A fatal error occurred: Failed to connect to Espressif device: No serial data received.
For troubleshooting steps visit: https://docs.espressif.com/projects/esptool/en/latest/troubleshooting.html
[21:58]```
##I also tried with a lower baud rate of 9600 and it was the same

**Comment by GiannisDorokidis** [comment]
There are some commends it the code i couldnt figure out how to make it seperate

**Reply by sfromis** [reply]
**Absolutely** do not flash `tasmota-minimal.bin` as first binary. That would have soft-bricked the device, and it even looks like you did extra work to get to have that wrong file.

Anyway, you did not get that far, it looks like you did not have a good/correct connection to the board, or not got it in flashing mode. Maybe your USB/serial adapter is not great, we do have a recommendation, and guide: https://tasmota.github.io/docs/Getting-Started/

Also, do not make that separate erase step, then you'd have to power off and get it in flashing mode again.

And do not just say "I followed some random guide", that tells zero about what you actually did.

**Reply by GiannisDorokidis** [reply]
It now worked and i have succesfuly writen the tasmota.bin with no problems from esptool. What i was doing was that sometimes i didnt put it into flash mode because it has an onboard led that tells you when you are on flash mode. Only problem with that is if you powercycle the device you still need to put it in flash mode the led stayed on so i thought it was already. 
Anyway thanks for the reply
